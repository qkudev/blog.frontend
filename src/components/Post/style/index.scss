.post {
  max-width: 600px;
  margin: 0 auto 3rem auto;

  &-header {
    a {
      font-size: 2rem;
    }
  }

  &-body {
    line-height: 1.5em;

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      //margin-top: 1.5em;
      margin: 0.5em 0;
      line-height: 1.25em;
    }

    p {
      margin-bottom: 1rem;
    }
  }
}
