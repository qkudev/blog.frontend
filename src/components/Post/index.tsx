import * as React from 'react'
import Markdown from 'react-markdown'
import moment from 'moment'
import './style'

type Props = IPost

function date(date: string) {
  return moment(date).format('YYYY/MM/DD, hh:mm')
}

function Post({ body, createdAt }: Props) {
  return (
    <div className='post'>
      <Markdown className='post-body' source={body} />
      <div className='post-date'>{date(createdAt)}</div>
    </div>
  )
}

export default Post
