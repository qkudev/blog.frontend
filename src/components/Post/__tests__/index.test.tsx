import * as React from 'react'
import renderer from 'react-test-renderer'
import Post from '../index'

const mockPost: IPost = {
  _id: 'qwerty',
  body: 'Some text',
  updatedAt: '2019-04-14T23:41:36.340Z',
  createdAt: '2019-04-14T23:41:36.340Z',
  __v: 0
}

describe('Component:Post', function() {
  test('Matches with snapshot', function() {
    const post = renderer.create(<Post {...mockPost} />)
    let tree = post.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
