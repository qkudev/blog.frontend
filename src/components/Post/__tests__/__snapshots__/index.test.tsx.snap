// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Post Matches with snapshot 1`] = `
<div
  className="post"
>
  <div
    className="post-body"
  >
    <p>
      Some text
    </p>
  </div>
  <div
    className="post-date"
  >
    2019/04/15, 06:41
  </div>
</div>
`;
